Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,236 @@
+
+                if (highestLastBalance < currentBalanceHolderFinalBalance)
+                {
+                    highestLastBalanceHolder = currentBalanceHolder;
+                    holderCount = 1;
+                }
+
+                if (highestLastBalance == currentBalanceHolderFinalBalance &&
+                    highestLastBalanceHolder != currentBalanceHolder &&
+                    holderCount == 2)
+                {
+                    highestLastBalanceHolder += ($" and {currentBalanceHolder}");
+                }
+                if (highestLastBalance == currentBalanceHolderFinalBalance &&
+                   highestLastBalanceHolder != currentBalanceHolder &&
+                   holderCount == 1)
+                {
+                    highestLastBalanceHolder += ($", {currentBalanceHolder}");
+                    holderCount++;
+                }
+                highestLastBalance = currentBalanceHolderFinalBalance > highestLastBalance ?
+                    currentBalanceHolderFinalBalance : highestLastBalance;
+            }
+            string stringHighesLasttBalance = highestLastBalance.ToString();
+
+            return holderCount == 1 ? 
+                Invariant($"{highestLastBalanceHolder} is the richest person. ¤{stringHighesLasttBalance}.") :
+                Invariant($"{highestLastBalanceHolder} are the richest people. ¤{stringHighesLasttBalance}.");
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            {
+                if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+                {
+                    return "N/A.";
+                }
+                int holderCount = 0;
+                double overallLowestLastBalance = double.MaxValue;
+                string currentBalanceHolder;
+                string overallLowestLastBalanceHolder = "";
+                bool isNegativeBalance = false;
+
+                foreach (string entry in peopleAndBalances)
+                {
+                    string[] userHistory = entry.Split(',');
+
+                    currentBalanceHolder = userHistory[0];
+
+                    double currentBalanceHolderFinalBalance = double.Parse(userHistory[^1]);
+
+                    if (overallLowestLastBalance > currentBalanceHolderFinalBalance)
+                    {
+                        overallLowestLastBalanceHolder = currentBalanceHolder;
+                        holderCount = 1;
+                    }
+
+                    if (overallLowestLastBalance == currentBalanceHolderFinalBalance &&
+                        overallLowestLastBalanceHolder != currentBalanceHolder &&
+                        holderCount == 2)
+                    {
+                        overallLowestLastBalanceHolder += ($" and {currentBalanceHolder}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4953253)